subsys/bluetooth/shell/bt.c

```diff
@@ -145,6 +145,15 @@ static struct bt_scan_filter {
 	bool pa_interval_set;
 } scan_filter;
 
+static bool scan_verbose_output;
+static char *scan_response_type_txt[] = {
+	"ADV_IND",
+	"ADV_DIRECT_IND",
+	"ADV_SCAN_IND",
+	"ADV_NONCONN_IND",
+	"SCAN_RSP",
+	"EXT_ADV"
+};
```

Comment:
This is an easy way of doing it, but this may break if any of the values change. While it is unlikely that they will change (given the current 1:1 between them and the HCI defined values), a more future proof solution would be something like `bt_tbs_state_str` that takes the type as input and uses a switch to return the string value. That way, if the value changes (e.g. if `BT_GAP_ADV_TYPE_ADV_DIRECT_IND` becomes 3 instead of 2) then it will continue to work. Furthermore, if a new type is added in HCI, then this would end up in a "out of bounds" runtime issue.

Reply:
So, the reasoning is, that it is highly unlikely that they are going to be changed, but I did have my thought of indexing into the array like that. However, a slick-fix would be to use:
`scan_response_type_txt[MIN(info->adv_type,
					   sizeof(scan_response_type_txt)/sizeof(scan_response_type_txt[0]) - 1)]);`
instead of an elaborate function for it. Would that be acceptable?

Reply:
I agree with @Thalley. What we want is this:

```suggestion
static char *scan_response_type_txt(enum bt_gap_adv_type arg)
{
	switch (arg) {
	case BT_GAP_ADV_TYPE_ADV_IND:
		return "ADV_IND";
	case BT_GAP_ADV_TYPE_ADV_DIRECT_IND:
		return "ADV_DIRECT_IND";
	case BT_GAP_ADV_TYPE_ADV_SCAN_IND:
		return "ADV_SCAN_IND";
	case BT_GAP_ADV_TYPE_ADV_NONCONN_IND:
		return "ADV_NONCONN_IND";
	case BT_GAP_ADV_TYPE_SCAN_RSP:
		return "SCAN_RSP";
	case BT_GAP_ADV_TYPE_EXT_ADV:
		return "EXT_ADV";
	default:
		return "unknown";
	}
}
```